payroll/models/hr_payslip.py

```diff
@@ -618,7 +618,10 @@ def get_lines_dict(self):
                 )
                 # set up localdict with current contract and employee values
                 localdict = dict(
-                    baselocaldict, employee=contract.employee_id, contract=contract
+                    baselocaldict,
+                    employee=contract.employee_id,
+                    contract=contract,
+                    payslip=payslip,
```

Comment:
Why are you adding payslip=payslip here? It's already provided by baselocaldict. See comment above.

Reply:
That's the whole reason, "payslip" object was never in the localdict, it was the Payslips (browsable class) object that was in it. 

In the practice, you can access the same fields, with the difference that it's not the real payslip object so if you want to call any not stored field from the previous "payslip" object, you can't. Now with this change, you can, and also continue having the same fields so we are no breaking any rules. 

Now we have a new object called "payslips" that refer to the browsable class. But "payslip" object was never the real object. I realised now, and i think it was a bug that has passed uncaught because the browsable class also retrieves the stored values so, in the most of the cases, you didn't realize that you are no using the real payslip object. 

I tested and it has no effect in rules using "payslip".